File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = [[int(el) for el in line.split()]
                       for line in matrix_string.splitlines()]

    def row(self, index):
        return self.matrix[index-1]

    def column(self, index):
        return [row[index-1] for row in self.matrix]

File: java/matrix/src/main/java/Matrix.java
import java.util.Arrays;

class Matrix {

    private final int[][] matrix;

    Matrix(String matrixAsString) {
        matrix = matrixAsString.lines()
                .map(line -> Arrays.stream(line.split("\\s+")).mapToInt(Integer::parseInt).toArray())
                .toArray(int[][]::new);
    }

    int[] getRow(int rowNumber) {
        return matrix[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        return Arrays.stream(matrix).mapToInt(row -> row[columnNumber - 1]).toArray();
    }
}
